src/drivers/angular2.js

```diff
@@ -0,0 +1,99 @@
+import { omit, replaceObject } from '../lib';
+/* eslint-disable new-cap, object-shorthand */
+
+
+export let angular2 = {
+
+    global() {
+        return (window.ng && window.ng.core && window.ng.platformBrowser) ? 
+        {
+            Component: window.ng.core.Component,
+            NgModule: window.ng.core.NgModule,
+            ElementRef: window.ng.core.ElementRef,
+            NgZone: window.ng.core.NgZone,
+            BrowserModule: window.ng.platformBrowser.BrowserModule
+        } 
+        : false;
+    },
+
+    register(xcomponent, { Component, NgModule, ElementRef, NgZone, BrowserModule }) {
+
+        xcomponent.log('initializing angular2 component');
+        
+        const getBindingMetadata = () => {
+            const inputs = [];
+            if (xcomponent.looseProps) {
+                inputs.push('props');
+            } else {
+                for (let key of Object.keys(xcomponent.props)) {
+                    inputs.push(key);
+                }
+            }
+            return { inputs };
+        };
+
+
+        function getProps(component) {
+            const props = component.props || omit(component, ['$xContext']);
+            return replaceObject(props, (value, key, fullKey) => {
+                if (typeof value === 'function') {
+                    return function () {
+                        let result;
+                        component.$xContext.zone.run(() => {
+                            result = value.apply(this, arguments);
+                        });
+                        return result;
+                    };
+                }
+            });
+        }
+
+        const bindingMetadata = getBindingMetadata();
+
+        const Angular2Component = 
+            Component({
+                selector: xcomponent.tag,
+                template: `
+                    <div></div>
+                `,
+                inputs: bindingMetadata.inputs
+            })
+            .Class({
+                constructor: [ElementRef, NgZone, function(elementRef, zone) { 
+                    this.$xContext = {
+                        elementRef,
+                        zone
+                    };
+                }],
+                ngOnInit: function () {
+                    const targetElement = this.$xContext.elementRef.nativeElement;
```

Comment:
Hmm, from a little reading I guess the recommended pattern is just to do all of this in the constructor.

In that case, can we use the `__XCOMPONENT__` constant instead of `$xContent`? So

```javascript
this[__XCOMPONENT__] = { element, zone };
```